### PR TITLE
Adding metrics for external Plugins - Hive, Batch and Presto

### PR DESCRIPTION
**Pull request by anandswaminathan**
Add metrics to all the external plugins.
In this PR,
- Resourcemanager interaction metrics for Hive and Presto.
- AWS Batch tasks metrics 

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

### CODE REVIEW

File: go/tasks/plugins/hive/executor.go [hunk 1]
```diff
@@ -150,7 +150,7 @@ func NewQuboleHiveExecutor(ctx context.Context, cfg *config.Config, quboleClient
 	return QuboleHiveExecutor{
 		id:              quboleHiveExecutorID,
 		cfg:             cfg,
-		metrics:         getQuboleHiveExecutorMetrics(scope),
+		metrics:         getQuboleHiveExecutorMetrics(scope.NewSubScope("hive")),
```
Comment (EngHabu) [inline go/tasks/plugins/hive/executor.go:153]: Reminder to make sure non of the existing alerts are based on the renamed metric or to fix them if they are

File: go/tasks/plugins/hive/executor_metrics.go [hunk 2]
```diff
@@ -21,11 +22,13 @@ var (
 func getQuboleHiveExecutorMetrics(scope promutils.Scope) QuboleHiveExecutorMetrics {
 	return QuboleHiveExecutorMetrics{
 		Scope: scope,
-		ReleaseResourceFailed: labeled.NewCounter("released_resource_failed",
+		ResourceReleased: labeled.NewCounter("resource_release_success",
+			"Resource allocation token released", scope),
```
Comment (EngHabu) [inline go/tasks/plugins/hive/executor_metrics.go:26]:
```suggestion
			"Resource allocation token released", scope, labeled.EmitUnlabeledMetric),
```

and same for the other ones

File: go/tasks/plugins/array/awsbatch/monitor.go [hunk 3]
```diff
@@ -64,11 +72,13 @@ func CheckSubTasksState(ctx context.Context, taskMeta core.TaskExecutionMetadata
 		Summary:  arraystatus.ArraySummary{},
 		Detailed: arrayCore.NewPhasesCompactArray(uint(currentState.GetExecutionArraySize())),
 	}
-
+	totalSuccesses := int64(0)
```
[the remark this answers is not in the document]
Comment (anandswaminathan) [reply go/tasks/plugins/array/awsbatch/monitor.go:75]: Awesome. That's why wanted your 👁️

File: go/tasks/plugins/hive/executor.go (hunk 1 above)
Comment (anandswaminathan) [reply go/tasks/plugins/hive/executor.go:153]: The metrics below were never added. It is added in this PR.

File: go/tasks/plugins/array/awsbatch/executor_metrics.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	SubTasksFailed     labeled.Counter
+	BatchJobTerminated labeled.Counter
+}
+
+func getAwsBatchExecutorMetrics(scope promutils.Scope) ExecutorMetrics {
+	return ExecutorMetrics{
+		Scope: scope,
+		SubTasksSubmitted: labeled.NewCounter("sub_task_submitted",
+			"Sub tasks submitted", scope),
+		SubTasksSucceeded: labeled.NewCounter("batch_task_success",
```
Comment (EngHabu) [inline go/tasks/plugins/array/awsbatch/executor_metrics.go:21]:
```suggestion
		SubTasksSucceeded: labeled.NewCounter("sub_task_success",
```

:)

File: go/tasks/plugins/hive/executor.go (hunk 1 above)
Comment (EngHabu) [reply go/tasks/plugins/hive/executor.go:153]: This will affect the existing ones tooThe resourceWaitTime

**Comment by anandswaminathan** [comment]
Ensured that `resource_wait_time` is not used anywhere. Merging this

**Pull request merged**